From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: William Douglas <william.douglas@intel.com>
Date: Wed, 23 May 2018 21:33:01 +0000
Subject: [PATCH] Remove -Werror from CFLAGS

---
 libopeniscsiusr/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libopeniscsiusr/Makefile b/libopeniscsiusr/Makefile
index 06b09f5..3983ed2 100644
--- a/libopeniscsiusr/Makefile
+++ b/libopeniscsiusr/Makefile
@@ -45,7 +45,7 @@ EXTRA_MAN_FILES = libopeniscsiusr.h.3
 OBJS = context.o misc.o session.o sysfs.o iface.o idbm.o node.o default.o
 
 CFLAGS ?= -O2 -g
-CFLAGS += -Wall -Werror -Wextra -fvisibility=hidden -fPIC
+CFLAGS += -Wall -Wextra -fvisibility=hidden -fPIC
 CFLAGS += $(shell $(PKG_CONFIG) --cflags libkmod)
 
 LDFLAGS += $(shell $(PKG_CONFIG) --libs libkmod)
