From 823b7ab54352cb78be3b22b1092e27947e3e8f6e Mon Sep 17 00:00:00 2001
From: William Douglas <william.douglas@intel.com>
Date: Wed, 23 May 2018 21:33:01 +0000
Subject: [PATCH 2/3] Remove -Werror from CFLAGS

---
 libopeniscsiusr/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libopeniscsiusr/Makefile b/libopeniscsiusr/Makefile
index 8b9b523..4a91fc6 100644
--- a/libopeniscsiusr/Makefile
+++ b/libopeniscsiusr/Makefile
@@ -42,7 +42,7 @@ EXTRA_MAN_FILES = libopeniscsiusr.h.3
 OBJS = context.o misc.o session.o sysfs.o iface.o
 
 CFLAGS ?= -O2 -g
-CFLAGS += -Wall -Werror -Wextra -fvisibility=hidden -fPIC
+CFLAGS += -Wall -Wextra -fvisibility=hidden -fPIC
 
 LIBADD =
 
-- 
2.17.0

