From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Patrick McCarty <patrick.mccarty@intel.com>
Date: Mon, 16 May 2022 11:23:02 -0700
Subject: [PATCH] Install generators with 755 perms, but not unit files

Signed-off-by: Patrick McCarty <patrick.mccarty@intel.com>
---
 etc/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/etc/Makefile b/etc/Makefile
index b3dbefc..eb569d4 100644
--- a/etc/Makefile
+++ b/etc/Makefile
@@ -82,10 +82,10 @@ install_systemd_service_files: $(DESTDIR)$(systemddir)/system $(SYSTEMD_DEST_SVC
 install_systemd_generator_files: $(DESTDIR)$(systemddir)/system-generators $(SYSTEMD_DEST_GEN_FILES)
 
 $(SYSTEMD_DEST_SVC_FILES): $(DESTDIR)$(systemddir)/system/%: systemd/%
-	$(INSTALL) $? $@
+	$(INSTALL) -m 644 $? $@
 
 $(SYSTEMD_DEST_GEN_FILES): $(DESTDIR)$(systemddir)/system-generators/%: systemd/%
-	$(INSTALL) $? $@
+	$(INSTALL) -m 755 $? $@
 
 install_iname: $(DESTDIR)$(HOMEDIR) $(ISCSI_INAME)
 	if [ ! -f $(INAME_DEST_FILE) ]; then \
