Index: open-iscsi-2.0-873/utils/iscsi-gen-initiatorname
===================================================================
--- open-iscsi-2.0-873.orig/utils/iscsi-gen-initiatorname
+++ open-iscsi-2.0-873/utils/iscsi-gen-initiatorname
@@ -22,6 +22,8 @@ if [ -d /sys/firmware/ibft/initiator ] ;
     read iSCSI_INITIATOR_NAME < /sys/firmware/ibft/initiator/initiator-name
 fi
 
+mkdir -p /etc/iscsi
+
 if [ -f /etc/iscsi/initiatorname.iscsi -a -z "$FORCE" ] ; then
     if [ "$iSCSI_INITIATOR_NAME" ] ; then
 	eval $(cat /etc/iscsi/initiatorname.iscsi | sed -e '/^#/d')
@@ -65,8 +67,7 @@ if [ ! -f /etc/iscsi/initiatorname.iscsi
 ## may reject this initiator.  The InitiatorName must be unique
 ## for each iSCSI initiator.  Do NOT duplicate iSCSI InitiatorNames.
 EOF
-	ISSUEDATE="1996-04"
-	INAME=$(/sbin/iscsi-iname -p iqn.$ISSUEDATE.de.suse:01)
+	INAME=$(/sbin/iscsi-iname)
 	printf "InitiatorName=$INAME\n"  >>/etc/iscsi/initiatorname.iscsi
 	chmod 0600 /etc/iscsi/initiatorname.iscsi
 fi
Index: open-iscsi-2.0-873/etc/systemd/iscsi-gen-initiatorname.service
===================================================================
--- /dev/null
+++ open-iscsi-2.0-873/etc/systemd/iscsi-gen-initiatorname.service
@@ -0,0 +1,9 @@
+[Unit]
+Description=iSCSI InitiatorName Generator
+ConditionPathExists=!/etc/iscsi/initiatorname.iscsi
+
+[Service]
+ExecStart=/usr/bin/iscsi-gen-initiatorname
+Type=oneshot
+RemainAfterExit=yes
+
Index: open-iscsi-2.0-873/etc/systemd/iscsid.service
===================================================================
--- open-iscsi-2.0-873.orig/etc/systemd/iscsid.service
+++ open-iscsi-2.0-873/etc/systemd/iscsid.service
@@ -1,13 +1,14 @@
 [Unit]
 Description=Open-iSCSI
 Documentation=man:iscsid(8) man:iscsiuio(8) man:iscsiadm(8)
-After=network.target NetworkManager-wait-online.service iscsiuio.service tgtd.service targetcli.service
+Wants=network-online.target iscsi-gen-initiatorname.service
+After=network.target network-online.target iscsi-gen-initiatorname.service iscsiuio.service tgtd.service targetcli.service
 
 [Service]
 Type=forking
 PIDFile=/var/run/iscsid.pid
-ExecStart=/usr/sbin/iscsid
-ExecStop=/sbin/iscsiadm -k 0 2
+ExecStart=/usr/bin/iscsid
+ExecStop=/usr/bin/iscsiadm -k 0 2
 
 [Install]
 WantedBy=multi-user.target
